src/Propel/Generator/Manager/MigrationManager.php

```diff
@@ -402,72 +404,57 @@ public function getMigrationObject($timestamp)
-
-    public function preUp(MigrationManager \$manager)
-    {
-        // add the pre-migration code here
-    }
-
-    public function postUp(MigrationManager \$manager)
-    {
-        // add the post-migration code here
-    }
-
-    public function preDown(MigrationManager \$manager)
-    {
-        // add the pre-migration code here
-    }
-
-    public function postDown(MigrationManager \$manager)
-    {
-        // add the post-migration code here
+        $connectionToVariableName = self::buildConnectionToVariableNameMap($migrationsUp, $migrationsDown);
+
+        $vars = [
+            'timestamp' => $timestamp,
+            'commentString' => addcslashes($comment, ','),
+            'suffix' => $suffix,
+            'timeInWords' => date('Y-m-d H:i:s', $timestamp),
+            'migrationAuthor' => ($author = $this->getUser()) ? 'by ' . $author : '',
+            'migrationClassName' => $this->getMigrationClassName($timestamp, $suffix),
+            'migrationsUp' => $migrationsUp,
+            'migrationsDown' => $migrationsDown,
+            'connectionToVariableName' => $connectionToVariableName,
+        ];
+
+        $template = new PropelTemplate();
+        $filePath = implode(DIRECTORY_SEPARATOR, [__DIR__, 'templates', 'migration_template.php']);
+        $template->setTemplateFile($filePath);
+
+        return $template->render($vars);
     }
 
     /**
-     * Get the SQL statements for the Up migration
+     *  * Builds an array mapping connection names to a string that can be used as a php variable name.
      *
-     * @return array list of the SQL strings to execute for the Up migration
-     *               the keys being the datasources
-     */
-    public function getUpSQL()
-    {
-        return $migrationUpString;
-    }
-
-    /**
-     * Get the SQL statements for the Down migration
+     * @param array $migrationsUp
+     * @param array $migrationsDown
      *
-     * @return array list of the SQL strings to execute for the Down migration
-     *               the keys being the datasources
+     * @return array
      */
-    public function getDownSQL()
+    protected static function buildConnectionToVariableNameMap(array $migrationsUp, array $migrationsDown): array
     {
-        return $migrationDownString;
-    }
-
-}
-EOP;
+        $connectionToVariableName = [];
+        foreach ([$migrationsUp, $migrationsDown] as $migrations) {
+            $connectionNames = array_keys($migrations);
+            foreach ($connectionNames as $index => $connectionName) {
+                if (array_key_exists($connectionName, $connectionToVariableName)) {
+                    continue;
+                }
+                $alphNums = preg_replace('/\W/', '', $connectionName);
+                if (strlen($alphNums) === 0) {
+                    $alphNums = $index;
+                }
+                $variableName = '$connection_' . $alphNums;
+                while (in_array($variableName, $connectionToVariableName)) {
```

Comment:
can we use 3rd arg `true` here please?

Reply:
Do you mean `in_array($variableName, $connectionToVariableName, true)`?
If so, why do you prefer it? Both `$variableName` and `$connectionToVariableName` are guaranteed to be strings and not null, so I am wondering about the upside of checking against type.

Reply:
Yes, this helps to communicate this.

Reply:
Oh, didn't think you'd respond, so I changed my comment. But yes, makes sense to use strict type checks here. Thanks for the tip!